source/common/event/file_event_impl.cc

```diff
@@ -114,7 +111,9 @@ void FileEventImpl::unregisterEventIfEmulatedEdge(uint32_t event) {
     ASSERT((event & (FileReadyType::Read | FileReadyType::Write)) == event);
     if (trigger_ == FileTriggerType::EmulatedEdge) {
       auto new_event_mask = enabled_events_ & ~event;
-      updateEvents(new_event_mask);
+      if (new_event_mask != enabled_events_) {
```

Comment:
comment here and below  why this case doesn't need the update?

Reply:
Writing the comment for updateEvents made me rethink what we should do here.  I reverted these changes and instead added a trigger_mode_ check to updateEvents so the update is skipped in modes where it is truly a no-op.